### ISSUE TITLE
allowed_sources being ignored

### ISSUE THREAD

**Issue by luison** [thumbor/thumbor#1380]
Having issues with allowed_domains ignoring our configuration and returning images from any unauthorised url.

### Expected behaviour
https://our.thumborserver.org/t/1925x800/smart/static.externaldomain.com/img/arc_25699_g.jpg
(not real url)
should not be returning an image

### Actual behaviour
any domain seems accepted by thumbor even if not included in allowed_sources

ALLOWED_SOURCES = ['.+.domain1.com', '.+.domain2.net', '.+.domain3.es']

### Operating system
thumbor vo 6.5 on LXC debian container behind a nginx reverse proxy

### Your thumbor.conf
Only including non-default public values:
```
MAX_WIDTH = 2400
MAX_HEIGHT = 3000
MAX_PIXELS = 35000000.0

#ALLOWED_SOURCES = #    [
#    ]
ALLOWED_SOURCES = ['.+.domain1.com', '.+.domain2.net', '.+.domain3.es']
QUALITY = 75

AUTO_WEBP = True

RESULT_STORAGE = 'thumbor.result_storages.file_storage'

LOADER = 'thumbor.loaders.file_loader_http_fallback'

FILE_LOADER_ROOT_PATH = '/home/webs-static/thumbor'


STORAGE_EXPIRATION_SECONDS = 300

FILE_STORAGE_ROOT_PATH = '/cache/thumbor'


DETECTORS = ["thumbor.detectors.feature_detector"]

OPTIMIZERS = [
  'thumbor.optimizers.jpegtran'
]

## Path for the jpegtran binary
## Defaults to: '/usr/bin/jpegtran'
JPEGTRAN_PATH = '/usr/bin/jpegtran'

FILTERS =     [
        'thumbor.filters.autojpg',
        'thumbor.filters.brightness',
        'thumbor.filters.colorize',
        'thumbor.filters.contrast',
        'thumbor.filters.rgb',
        'thumbor.filters.round_corner',
        'thumbor.filters.quality',
        'thumbor.filters.noise',
        'thumbor.filters.watermark',
        'thumbor.filters.equalize',
        'thumbor.filters.fill',
        'thumbor.filters.sharpen',
        'thumbor.filters.strip_exif',
        'thumbor.filters.strip_icc',
        'thumbor.filters.frame',
        'thumbor.filters.grayscale',
        'thumbor.filters.rotate',
        'thumbor.filters.format',
        'thumbor.filters.max_bytes',
        'thumbor.filters.convolution',
        'thumbor.filters.blur',
        'thumbor.filters.extract_focal',
        'thumbor.filters.focal',
        'thumbor.filters.no_upscale',
        'thumbor.filters.saturation',
        'thumbor.filters.max_age',
        'thumbor.filters.curve',
        'thumbor.filters.background_color',
        'thumbor.filters.upscale',
        'thumbor.filters.proportion',
        'thumbor.filters.stretch',
    ]

RESULT_STORAGE_EXPIRATION_SECONDS = 7200
RESULT_STORAGE_FILE_STORAGE_ROOT_PATH = '/cache/thumbor/results'

RESULT_STORAGE_STORES_UNSAFE = True

SENTRY_DSN_URL = 'http://xxxxxxxxxxxxxxxxxx@mon.domain.es:9000/6'

COMMUNITY_EXTENSIONS = [
    'tc_purger'
]
```

**Comment by mvhirsch** [comment]
I think you missed up a `.` (dot) in your configuration. It's a regex, so your `ALLOWED_SOURCES` should look like this:

```
# please notice the backslash `\` right before the `.domain`
ALLOWED_SOURCES = ['.+\.domain1.com', '.+\.domain2.net', '.+\.domain3.es']
```

----

Is it possible to see a non-redacted version of your allowed sources (and actual "externaldomain.com" example), to help you better?

**Comment by luison** [comment]
Thanks but does not seem to solve the issue.
I've updated from our config the loader which was not included:
LOADER = 'thumbor.loaders.file_loader_http_fallback'

We've tracked that if we use the http_loader the filter works fine but as we use the "file_loader_http_fallback" for local files first, in case the file does not exists, in the fallback to http the validation of allowed sources seems ignored. Not sure if the need to be defined in another form.

**Comment by heynemann** [comment]
This seems to be a bug. Sorry for that. You can either wait for a fix, or you can create your own loader. Thanks for reporting!

**Comment by heynemann** [comment]
Released under 7.0.4. Please reopen if it is still an issue.

**Comment by heynemann** [comment]
Actually released under 7.0.5 :)
